GUI/SearchTaskWindow.xaml.cs

```diff
@@ -1017,6 +906,139 @@ private static Proteomics.SilacLabel ConvertSilacDataGridInfoToSilacLabel(SilacI
+                }
+                if (!addCompIonCheckBox.IsChecked.Value)
+                {
+                    MessageBox.Show("Warning: Complementary ions are strongly recommended when using this algorithm.");
+                }
+                //only use N or C termini, not both
+                if (cTerminalIons.IsChecked.Value)
+                {
+                    nTerminalIons.IsChecked = false;
+                }
+                else
+                {
+                    nTerminalIons.IsChecked = true;
+                }
+            }
+        }
+
+        private FragmentationTerminus GetFragmentationTerminus()
+        {
+            if (nTerminalIons.IsChecked.Value && !cTerminalIons.IsChecked.Value)
+            {
+                return FragmentationTerminus.N;
+            }
+            else if (!nTerminalIons.IsChecked.Value && cTerminalIons.IsChecked.Value)
+            {
+                return FragmentationTerminus.C;
+            }
+            else if (!nTerminalIons.IsChecked.Value && !cTerminalIons.IsChecked.Value) //why would you want this
+            {
+                MessageBox.Show("Warning: No ion types were selected. MetaMorpheus will be unable to search MS/MS spectra.");
+                return FragmentationTerminus.None;
+            }
+            else
+            {
+                return FragmentationTerminus.Both;
+            }
+        }
+
+        //string out is for error messages
+        private void SilacUpdates(out string error)
+        {
+            error = "";
+            if (StaticSilacLabelsObservableCollection.Count != 0)
+            {
+                //remove the unlabeled
+                for (int i = 0; i < StaticSilacLabelsObservableCollection.Count; i++)
+                {
+                    if (StaticSilacLabelsObservableCollection[i].SilacLabel == null)
+                    {
+                        StaticSilacLabelsObservableCollection.RemoveAt(i);
+                        break;
+                    }
+                }
+
+                //Validate it really quick to determine if they're all multiplex (normal), or if it's a turnover experiment (requires one start label and one end label, either of which may be unlabeled)
+                //if they're all multiplex
+                if (StaticSilacLabelsObservableCollection.All(x => x.LabelType == SilacModificationWindow.ExperimentType.Multiplex))
+                {
+                    List<Proteomics.SilacLabel> labelsToSave = new List<Proteomics.SilacLabel>();
+                    foreach (SilacInfoForDataGrid info in StaticSilacLabelsObservableCollection)
+                    {
+                        labelsToSave.Add(ConvertSilacDataGridInfoToSilacLabel(info));
+                    }
+                    TheTask.SearchParameters.SilacLabels = labelsToSave;
+                    //Clear existing start and end labels, if any
+                    TheTask.SearchParameters.StartTurnoverLabel = null;
```

Comment:
Here's the GUI bug fix. Setting these to null if it's a multiplex to overwrite previous entries